apps/root/src/common/components/form-wallet-selector/index.tsx

```diff
@@ -0,0 +1,110 @@
+import useAccountService from '@hooks/useAccountService';
+import useTrackEvent from '@hooks/useTrackEvent';
+import useWallets from '@hooks/useWallets';
+import React from 'react';
+import { useIntl } from 'react-intl';
+import { Chip, colors, ContainerBox, Select, Skeleton, Typography, WalletIcon } from 'ui-library';
+import Address from '../address';
+import { formatUsdAmount } from '@common/utils/currency';
+import { useWalletUsdBalances } from '@state/balances/hooks';
+import useSelectedNetwork from '@hooks/useSelectedNetwork';
+import useActiveWallet from '@hooks/useActiveWallet';
+import { find, orderBy } from 'lodash';
+import { Wallet } from 'common-types';
+import { formatWalletLabel } from '@common/utils/parsing';
+
+type OptionWithKey = {
+  key: string;
+  wallet: Wallet;
+  usdBalance?: number;
+  isLoading: boolean;
+};
+
+const WalletItem = ({
+  item: { wallet, usdBalance, isLoading },
+  isRenderValue,
+}: {
+  item: OptionWithKey;
+  isRenderValue?: boolean;
+}) => {
+  const intl = useIntl();
+  const { address, label, ens } = wallet;
+  const { primaryLabel, secondaryLabel } = formatWalletLabel(address, label, ens);
+  return (
+    <ContainerBox alignItems="center" justifyContent="space-between" key={wallet.address} flex={1} gap={3}>
+      <ContainerBox alignItems="center" flex={1} gap={3}>
+        <WalletIcon />
+        {isRenderValue ? (
```

Comment:
same as before, I really dont like having conditionals inside the components based on if they are rendering as a child or as an item, I think it would be better to have different components for each. WDYT?

Reply:
This was easier to implement because both (listItem and renderedItem) are very similar, but I'll give it a shot by having them in two different components